Mathlib/CategoryTheory/Galois/EssSurj.lean

```diff
@@ -0,0 +1,261 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.CategoryTheory.Galois.Full
+import Mathlib.CategoryTheory.Galois.Topology
+import Mathlib.Topology.Algebra.OpenSubgroup
+
+/-!
+
+# Essential surjectivity of fiber functors
+
+Let `F : C ⥤ FintypeCat` be a fiber functor of a Galois category `C` and denote by
+`H` the induced functor `C ⥤ Action FintypeCat (Aut F)`.
+
+In this file we show that the essential image of `H` consists of the finite `Aut F`-sets where
+the `Aut F` action is continuous.
+
+## Main results
+
+- `exists_lift_of_quotient_openSubgroup`: If `U` is an open subgroup of `Aut F`, then
+  there exists an object `X` such that `F.obj X` is isomorphic to `Aut F ⧸ U` as
+  `Aut F`-sets.
+- `exists_lift_of_continuous`: If `X` is a finite, discrete `Aut F`-set, then
+  there exists an object `A` such that `F.obj A` is isomorphic to `X` as
+  `Aut F`-sets.
+
+## Strategy
+
+We first show that every finite, discrete `Aut F`-set `Y` has a decomposition into connected
+components and each connected component is of the form `Aut F ⧸ U` for an open subgroup `U`.
+Since `H` preserves finite coproducts, it hence suffices to treat the case `Y = Aut F ⧸ U`.
+For the case `Y = Aut F ⧸ U` we closely follow the second part of Stacks Project Tag 0BN4.
+
+-/
+
```

Comment:
Just write `noncomputable theory` somewhere up here instead of all the repeated `noncomputable`?

Reply:
`noncomputable theory` does not seem to work, but I added `noncomputable section`. I hope that's what you meant.